jsdom-nodejs-env/src/main/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnv.scala

```diff
@@ -42,33 +44,41 @@ class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends JSEnv {
   def startWithCom(input: Input, runConfig: RunConfig,
       onMessage: String => Unit): JSComRun = {
     JSDOMNodeJSEnv.validator.validate(runConfig)
+    val validatedInput = validateInput(input)
     ComRun.start(runConfig, onMessage) { comLoader =>
-      val files = initFiles ::: (comLoader :: codeWithJSDOMContext(input))
-      internalStart(files, runConfig)
+      internalStart(comLoader :: Nil, validatedInput, runConfig)
     }
   }
 
-  private def internalStart(files: List[VirtualBinaryFile],
-      runConfig: RunConfig): JSRun = {
+  private def validateInput(input: Input): ValidatedInput = input match {
+    case Input.ScriptsToLoad(scripts) =>
+      ValidatedInput.ScriptsToLoad(scripts)
+    case _ =>
+      throw new UnsupportedInputException(input)
+  }
+
+  private def internalStart(initFiles: List[VirtualBinaryFile],
+      input: ValidatedInput, runConfig: RunConfig): JSRun = {
     val command = config.executable :: config.args
     val externalConfig = ExternalJSRun.Config()
       .withEnv(env)
       .withRunConfig(runConfig)
-    ExternalJSRun.start(command, externalConfig)(JSDOMNodeJSEnv.write(files))
+    ExternalJSRun.start(command, externalConfig)(
+        JSDOMNodeJSEnv.write(initFiles ::: codeWithJSDOMContext(input)))
```

Comment:
So, looking at this all, I don't understand why `codeWithJSDOMContext` moved down here. Because it seems it requires this whole `ValidatedInput` matchinery whereas before we could just pass the unvalidated input to `codeWithJSDOMContext` and it would fail if it is unhappy.

Reply:
hahahahaha *matchinery*, totally unintended :P

Reply:
Right, the refactorings are not really necessary on their own at this point. I did it this way because
* it follows the internal design of `NodeJSEnv`, so understanding one helps understanding the other and conversely
* we'll need the refactoring to support ECMAScript modules later, which is something we should do at some point, although we'll need https://github.com/jsdom/jsdom/issues/2475 to happen first

I can also cancel the whole refactoring and just call `scriptFiles` at the top instead of `validateInput`.

Reply:
Yes, please. Could you for now just remove `initFiles` from everywhere and do other refactorings later, once we actually need them?

At the moment it is very hard to understand why some things are done.